src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -449,61 +398,102 @@ public string OutputAssembly
-            }
-        }
-        internal OutputAssemblyType outputType = OutputAssemblyType.Library;
-        internal bool outputTypeSpecified = false;
+        public OutputAssemblyType OutputType { get; set; } = OutputAssemblyType.Library;
 
         /// <summary>
         /// Flag to pass the resulting types along.
         /// </summary>
-        [Parameter()]
+        [Parameter(ParameterSetName = FromSourceParameterSetName)]
+        [Parameter(ParameterSetName = FromMemberParameterSetName)]
+        [Parameter(ParameterSetName = FromPathParameterSetName)]
+        [Parameter(ParameterSetName = FromLiteralPathParameterSetName)]
+        [Parameter(ParameterSetName = FromAssemblyNameParameterSetName)]
         public SwitchParameter PassThru { get; set; }
 
+        /// <summary>
+        /// Flag to compile without loading compiled assembly.
+        /// </summary>
+        [Parameter(ParameterSetName = FromSourceParameterSetName)]
+        [Parameter(ParameterSetName = FromMemberParameterSetName)]
+        [Parameter(ParameterSetName = FromPathParameterSetName)]
+        [Parameter(ParameterSetName = FromLiteralPathParameterSetName)]
+        public SwitchParameter CompileOnly { get; set; }
+
         /// <summary>
         /// Flag to ignore warnings during compilation.
         /// </summary>
-        [Parameter()]
-        public SwitchParameter IgnoreWarnings { get; set; }
+        [Parameter(ParameterSetName = FromSourceParameterSetName)]
+        [Parameter(ParameterSetName = FromMemberParameterSetName)]
+        [Parameter(ParameterSetName = FromPathParameterSetName)]
+        [Parameter(ParameterSetName = FromLiteralPathParameterSetName)]
+        [Alias("IW", "IgnoreWarnings")]
+        public SwitchParameter WarnAsError { get; set; }
+
+        /// <summary>
+        /// Roslyn command line parameters.
+        /// https://github.com/dotnet/roslyn/blob/master/docs/compilers/CSharp/CommandLine.md
+        /// https://github.com/dotnet/roslyn/blob/master/docs/compilers/Visual%20Basic/CommandLine.md
+        ///
+        /// Parser options:
+        ///     langversion:string - language version from:
+        ///                                 [enum]::GetNames([Microsoft.CodeAnalysis.CSharp.LanguageVersion])
+        ///                                 [enum]::GetNames([Microsoft.CodeAnalysis.VisualBasic.LanguageVersion])
+        ///     define:symbol list - preprocessor symbols:
+        ///                                 /define:UNIX,DEBUG      - CSharp
+        ///                                 /define:UNIX=1,DEBUG=1  - VisualBasic
+        ///
+        /// Compilation options:
+        ///     optimize{+|-}            - optimization level
+        ///     parallel{+|-}            - concurrent build
+        ///     warnaserror{+|-}         - report warnings to errors
+        ///     warnaserror{+|-}:strings - report specific warnings to errors
+        ///     warn:number              - warning level (0-4) for CSharp
```

Comment:
I guess this applies to both csharp and vb. If that's the case, `for CSharp` should be removed.

Reply:
I don't found the option for VB https://github.com/dotnet/roslyn/blob/master/docs/compilers/Visual%20Basic/CommandLine.md

Reply:
OK, then never mind. #Closed.